File: infra/deploy-evolution-vm/test_sre.js
const axios = require('axios');
const fs = require('fs');
const path = require('path');

const BASE_URL = 'http://localhost:8080';
const INSTANCE_NAME = 'debug_final';

async function runTest() {
    console.log(`--- Iniciando Teste de Estresse (Node.js): ${INSTANCE_NAME} ---`);

    let API_KEY = '';
    try {
        const envContent = fs.readFileSync(path.join(process.env.HOME, 'deploy-evolution-vm/.env'), 'utf8');
        const match = envContent.match(/AUTHENTICATION_API_KEY=(.*)/);
        if (match) API_KEY = match[1].trim();
    } catch (e) {
        console.error('ERRO: Não foi possível ler o arquivo .env.');
        process.exit(1);
    }

    const headers = {
        'apikey': API_KEY,
        'Content-Type': 'application/json'
    };

    try {
        // 1. Criar
        console.log(`[1/3] Criando instância ${INSTANCE_NAME}...`);
        const createResp = await axios.post(`${BASE_URL}/instance/create`, {
            instanceName: INSTANCE_NAME,
            qrcode: true
        }, { headers });
        console.log(`Resposta Create: ${createResp.status}`);

        // 2. Delay
        console.log('[2/3] Aguardando 5 segundos para estabilização...');
        await new Promise(resolve => setTimeout(resolve, 5000));

        // 3. Connect
        console.log('[3/3] Recuperando QR Code via /instance/connect...');
        const connectResp = await axios.get(`${BASE_URL}/instance/connect/${INSTANCE_NAME}`, { headers });
        
        if (connectResp.status === 200 && connectResp.data.base64) {
            console.log('\nSUCESSO: QR CODE GERADO EM BASE64!');
        } else {
            console.log(`\nFALHA: Resposta inesperada (Status ${connectResp.status})`);
            console.log(JSON.stringify(connectResp.data, null, 2));
        }
    } catch (e) {
        console.error(`\nERRO: ${e.message}`);
        if (e.response) {
            console.error('Detalhes:', JSON.stringify(e.response.data, null, 2));
        }
    }
}

runTest();

File: infra/deploy-evolution-vm/test_sre.py
import axios
import time
import json
import sys

# Configurações do Teste (Ambiente Local da VM)
BASE_URL = "http://localhost:8080"
INSTANCE_NAME = "debug_final"

# Tenta ler a API Key do .env local se disponível
try:
    with open("deploy-evolution-vm/.env", "r") as f:
        for line in f:
            if line.startswith("AUTHENTICATION_API_KEY="):
                API_KEY = line.split("=")[1].strip()
                break
except:
    print("ERRO: Não foi possível ler o arquivo .env. Certifique-se de estar no diretório correto.")
    sys.exit(1)

HEADERS = {
    "apikey": API_KEY,
    "Content-Type": "application/json"
}

def run_test():
    print(f"--- Iniciando Teste de Estresse: {INSTANCE_NAME} ---")
    
    # 1. Criar Instância
    print(f"[1/3] Criando instância {INSTANCE_NAME}...")
    try:
        resp = axios.post(f"{BASE_URL}/instance/create", 
                         json={"instanceName": INSTANCE_NAME, "qrcode": True}, 
                         headers=HEADERS)
        print(f"Resposta Create: {resp.status_code}")
        # print(json.dumps(resp.json(), indent=2))
    except Exception as e:
        print(f"FALHA NO CREATE: {str(e)}")
        return

    # 2. Delay Técnico (Checklist SRE)
    print("[2/3] Aguardando 5 segundos para estabilização do socket...")
    time.sleep(5)

    # 3. Tentar recuperar QR Code
    print("[3/3] Recuperando QR Code via /instance/connect...")
    try:
        resp = axios.get(f"{BASE_URL}/instance/connect/{INSTANCE_NAME}", headers=HEADERS)
        data = resp.json()
        
        if resp.status_code == 200 and "base64" in data:
            print("\nSUCESSO: QR CODE GERADO EM BASE64!")
            # print(f"Preview: {data['base64'][:50]}...")
        else:
            print(f"\nFALHA: Resposta inesperada (Status {resp.status_code})")
            print(json.dumps(data, indent=2))
            
    except Exception as e:
        print(f"\nERRO CATASTRÓFICO: {str(e)}")
        if hasattr(e, 'response') and e.response:
            print(f"Detalhes Baileys: {e.response.text}")

if __name__ == "__main__":
    run_test()
